### ISSUE TITLE
Be louder about errors while updating feeds

### ISSUE THREAD

**Issue by jabranham** [skeeto/elfeed#196]
Hello - 

I noticed that after elfeed updated, there was a buffer with the following errors:

```
[2017-01-31 08:03:23] [error]: http://apr.sagepub.com/rss/current.xml: (error XML: (Not Well-Formed) Invalid entity reference)
[2017-01-31 08:03:23] [error]: http://prq.sagepub.com/rss/current.xml: (error XML: (Not Well-Formed) Invalid entity reference)
[2017-01-31 08:03:24] [error]: http://rap.sagepub.com/rss/current.xml: (error XML: (Not Well-Formed) Invalid entity reference)
[2017-01-31 08:03:24] [error]: http://spa.sagepub.com/rss/current.xml: (error XML: (Not Well-Formed) Invalid entity reference)
[2017-01-31 08:03:24] [error]: http://cps.sagepub.com/rss/: (error XML: (Not Well-Formed) Invalid entity reference)
```

Because apparently Sage moved their RSS feeds. I don't know how long ago this happened. 

It would be nice if elfeed was a bit louder about this type of error, as I didn't notice that the `*elfeed-log*` buffer was even being created until today.

**Comment by skeeto** [comment]
If you want something louder, add own definition of loud to these two
hooks:

* elfeed-http-error-hooks
* elfeed-parse-error-hooks

Long ago Elfeed used to be a lot louder about these things. Every time a
feed failed it would announce it in the minibuffer and in *Messages*.
However, if you're following 100+ feeds, failures are pretty much a
daily occurrence and it gets annoying pretty fast. The web is kind of a
fragile thing that way.

However, you make a good point. There is something to be said about
keeping a failure tally. Elfeed already keeps a ":failures" metadata
property on each feed to tally an absolute number of failures. Perhaps
it needs a separate tally of consecutive failures, cleared on success.
If that tally goes above some threshold, then Elfeed should make some
noise about it.

**Comment by jabranham** [comment]
Thanks. I'll look into those hooks.

What you suggest would definitely be helpful. Maybe instead of notifying about the number of failures, elfeed should keep information on ":last-successful-update" and notify if more than X units of time have passed? By default, perhaps a day or two?

**Comment by sjLambda** [comment]
Would you consider something like this a parse-error?

cond: Don’t know how to access metadata on [cl-struct-elfeed-feed "http://rss.slashdot.org/Slashdot/slashdot" "http://rss.slashdot.org/Slashdot/slashdot" "Slashdot" nil (:last-modified "Fri, 28 Apr 2017 04:22:19 GMT" :etag "IgEPrVr96xl44ewoYfCXvLSPYtI" :failures 19)]

I keep getting this for all my feeds, not just this one site. FYI, the xml is well-formed and don't see any obvious struct errors.

**Comment by skeeto** [comment]
@sjLambda, did you recently switch to the dev version of Emacs (#207)?
There's nothing wrong with that feed, and this is due to a database
format change. Emacs 26 will change the way structures are printed,
preventing the index file from migrating to version 26 from previous
versions. There's currently no solution to convert the index from the
old version to the new, so you'll have to purge it and start fresh.

**Comment by sjLambda** [comment]
Yes I use dev version most of the time. I can do the purging. Thanks.

Also thanks for elfeed, hard to believe I'm already hooked on it.

--S.J.
